### ISSUE TITLE
[require-await] Remove from recommended config to match eslint:recommended

### ISSUE THREAD

**Issue by jordanbtucker** [typescript-eslint/typescript-eslint#4283]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Additional Info**

ESLint does not enable `require-await` in its recommended configuration. Coming from JavaScript to TypeScript, it is strange to see `@typescript-eslint/require-await` enabled in this project's recommended configuration. Should this rule be removed from this project's recommended configuration to keep in line with ESLint?

Note: This is a duplicate of #4282. Please close/delete that one in favor of this one as I opened with with the wrong account.

**Comment by bradzacher** [comment]
Short answer: no.

Long answer:
First - we don't align with ESLint on everything. ESLint's recommended set is intentionally very, very conservative. Ours is too, but less so.

Second - ESLint's rules don't use type information. So there are many cases they will false-positive on.
This causes a number of eslint's rules to not be in the recommended config - because recommending a rule that false-positives would be an awful UX.

In particular - ESLint's `require-await` version does not use type information so it is not able to understand when a function is marked as `async` and also returns a promise:
```ts
async function foo() {
  return Promise.resolve();
}
```
There's also cases like async generators which ESLint's version can't handle.

Our version of the rule, OTOH, does use type information and can understand these patterns - thus there aren't false positives any more.

It's best-practice to not use `async` unless you have an asynchronous function, thus this rule is recommended.

**Comment by jordanbtucker** [comment]
Thank you for the thorough answer.
